application/models/services/Quotas.php

```diff
@@ -384,4 +392,212 @@ public function checkActionPermissions($action)
         }
         return $permissionOk;
     }
+
+    /**
+     * checkCompletedQuota() returns matched quotas information for the current response
+     * @param integer $surveyid Survey ID
+     * @param array $updatedValues The fields to be updated in the current request
+     * @param bool $return Set to true to return information or false to execute the quota and display a message
+     * @return array|void Nested array, Quotas->Members->Fields, includes quota information matched in session
+     * @throws \CException
+     * @throws \CHttpException
+     * @throws \Throwable
+     * @throws \WrongTemplateVersionException
+     */
+    static public function checkCompletedQuota(int $surveyid, array $updatedValues = [], bool $return = false)
+    {
+        /* Check if session is set */
+        if (!isset(App()->session['survey_' . $surveyid]['srid'])) {
+            return;
+        }
+        /* Check if Response is already submitted : only when "do" the quota: allow to send information about quota */
+        $oResponse = Response::model($surveyid)->findByPk(App()->session['survey_' . $surveyid]['srid']);
+        if (!$return && $oResponse && !is_null($oResponse->submitdate)) {
+            return;
+        }
+        // EM call 2 times quotas with 3 lines of php code, then use static.
+        static $aMatchedQuotas;
+        if (!$aMatchedQuotas) {
```

Comment:
Maybe musty check is_null here (not related to this PR, just more improvement)

Avoid this request : `$aQuotas = Quota::model()->with('languagesettings',
                'quotaMembers.question')->findAllByAttributes(['sid' => $surveyid]);`